internal/certs.go

```diff
@@ -0,0 +1,160 @@
+package internal
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"net"
+	"os"
+	"path"
+	"time"
+
+	"github.com/open-telemetry/opamp-go/protobufs"
+)
+
+func CreateClientTLSConfig(clientCert *tls.Certificate, certsDir string) (*tls.Config, error) {
+	// Read the CA's public key. This is the CA that signs the server's certificate.
+	caCertBytes, err := os.ReadFile(path.Join(certsDir, "certs/ca.cert.pem"))
```

Comment:
Since this is now a more general-purpose utility let's avoid making assumptions about directory structure here and instead just accept as a function parameter the full path to the pem file instead of accepting the `certsDir` as a parameter.
The caller should be responsible for knowing in which directory the file resides.